esmvalcore/preprocessor/_area.py

```diff
@@ -315,6 +315,25 @@ def _select_representative_point(shape, lon, lat):
     return select
 
 
+def _correct_coords_from_shapefile(cube, cmor_coords):
+    """Get correct lat and lon from shapefile."""
+    lon = cube.coord(axis='X').points
+    lat = cube.coord(axis='Y').points
+    if cube.coord(axis='X').ndim < 2:
+        lon, lat = np.meshgrid(lon, lat, copy=False)
+
+    if not cmor_coords:
+        # Wrap around longitude coordinate to match data
+        lon_180 = np.where(lon >= 180., lon - 360., lon)
+        # the NE mask has no points at x = -180 and y = +/-90
+        # so we will fool it and apply the mask at (-179, -89, 89) instead
+        lon = np.where(lon_180 == -180., lon_180 + 1., lon_180)
+        lat_0 = np.where(lat == -90., lat + 1., lat)
+        lat = np.where(lat_0 == 90., lat_0 - 1., lat_0)
+
```

Comment:
This is not needed for general shapefiles, maybe it only applies to natural earth masks? I think it should be removed.

Reply:
it's needed for NE shapefiles (yes, specifically NE as far as Ive seen) otherwise you end up with a white delta function or stripe in the middle of the map